apps/storefront/src/pages/AccountSetting/index.test.tsx

```diff
@@ -0,0 +1,157 @@
+    server.use(
+      graphql.query('B2BAccountFormFields', () =>
+        HttpResponse.json({
+          data: {
+            accountFormFields: [],
+          },
+        }),
+      ),
+      graphql.query('getB2bAccountSettings', () =>
+        HttpResponse.json({
+          data: {
+            accountSettings: {},
+          },
+        }),
+      ),
+      http.get('*', ({ request }) => {
+        console.error(`Unhandled GET request: ${request.url.toString()}`);
+        // You can return a specific error response or let it fall through (which MSW will warn about)
+        // For debugging, you might want to throw an error here to immediately fail the test.
+        // throw new Error(`Unhandled GET request: ${request.url.toString()}`);
+        return HttpResponse.json({ message: 'Unhandled Mocked GET Request' }, { status: 500 });
+      }),
+    );
+
+    renderWithProviders(<AccountSetting />, {
+      preloadedState: {
+        company: companyState,
+      },
+    });
+
+    await vi.waitFor(
+      async () => {
+        const bannerTitle = screen.queryByText('Upgrade to a business account');
+        expect(bannerTitle).not.toBeInTheDocument();
+      },
+      {
+        timeout: 500,
+      },
+    );
+  });
+
+  it('displays the B2B upgrade bannel when user is B2C user', async () => {
+    const customer = buildCustomerWith({
+      id: 123,
+      userType: UserTypes.MULTIPLE_B2C,
+    });
+
+    const companyInfoInCompanyState = {
+      id: '79',
+      companyName: 'b2bc',
+      status: CompanyStatus.INACTIVE,
+    };
+
+    const companyState = buildCompanyStateWith({
+      customer,
+      companyInfo: companyInfoInCompanyState,
+      permissions: [],
+    });
+
+    server.use(
+      graphql.query('B2BAccountFormFields', () =>
+        HttpResponse.json({
+          data: {
+            accountFormFields: [],
+          },
+        }),
+      ),
+      graphql.query('getB2cAccountSettings', () =>
+        HttpResponse.json({
+          data: {
+            customerAccountSettings: [],
+          },
+        }),
+      ),
+      http.all('*', ({ request }) => {
+        console.error(`Unhandled  request: ${request.url.toString()}`);
+        // You can return a specific error response or let it fall through (which MSW will warn about)
+        // For debugging, you might want to throw an error here to immediately fail the test.
+        // throw new Error(`Unhandled GET request: ${request.url.toString()}`);
+        return HttpResponse.json({ message: 'Unhandled Mocked GET Request' }, { status: 500 });
+      }),
+    );
+
+    renderWithProviders(<AccountSetting />, {
+      preloadedState: {
+        company: companyState,
+      },
+    });
+
+    await vi.waitFor(
+      async () => {
+        expect(screen.getByText('Upgrade to a business account')).toBeDefined();
+        const upgradeLink = screen.getByRole('link', { name: 'Upgrade' });
+        expect(upgradeLink).toHaveAttribute('href', '/registeredbctob2b');
```

Comment:
it is not recommended to have more than one expect per waitFor, and also, i don't think we need it, we can use `findByText` to get the alert and then find the button normally with `getBy*`

```tsx
// findBy* will wait for the element to appear
const alert = await screen.findByRole('alert', { name: /Upgrade to a business acount/ });

const upgradeLink = within(alert).getByRole('link', { name: 'Upgrade' };

expect(upgradeLink).toHaveAttribute('href', '/registeredbctob2b');
```